data/test/scenarios/wml_tests/ScenarioWML/EventWML/events_in_ability_and_special_tags.cfg

```diff
@@ -0,0 +1,140 @@
+#####
+# API(s) being tested: [unit][abilities][dummy][event]
+##
+# Actions:
+# Define a custom event in an (arbitrarily named) ability tag in a unit made with ActionWML
+# Fire the custom event
+##
+# Expected end state:
+# The custom event has executed
+#####
+{GENERIC_UNIT_TEST "event_in_ability_in_unit" (
+    [event]
+        name=start
+        [unit]
+            type=Elvish Lady
+            [abilities]
+                [dummy]
+                    id=ability_in_unit
+                    [event]
+                        name=event_in_ability_in_unit
+                        {VARIABLE pass_test 1}
+                    [/event]
+                [/dummy]
+            [/abilities]
+        [/unit]
+        {VARIABLE pass_test 0}
+        [fire_event]
+            name=event_in_ability_in_unit
+        [/fire_event]
+        {RETURN ({VARIABLE_CONDITIONAL pass_test equals 1})}
+    [/event]
+)}
+
+#####
+# API(s) being tested: [unit][specials][dummy][event]
+##
+# Actions:
+# Define a custom event in an (arbitrarily named) weapon special tag in a unit made with ActionWML
+# Fire the custom event
+##
+# Expected end state:
+# The custom event has executed
+#####
+{GENERIC_UNIT_TEST "event_in_special_in_unit" (
+    [event]
+        name=start
+        [unit]
+            type=Elvish Lady
+            [attack]
+                name,type,range,damage,number=stick,impact,melee,2,2
```

Comment:
I put these all on one line for my example to save spave, but it might be better to break it up a bit for clarity.